### PR TITLE
feat(switch): add new component switch

### PR DESCRIPTION
**Pull request by CarolineZow**
Signed-off-by: Caroline Zowczak <caroline.zowczak@corp.ovh.com>

### CODE REVIEW

File: packages/oui-switch/_mixins.less [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      &:focus ~ .oui-switch__slider {
+        box-shadow: inset 0 1px 1px 0 rgba(0, 0, 0, .25), 0px 0px 3px 1px rgba(0, 0, 0, 0.15);
+      }
+
+      &:checked {
+        ~ .@{switch-selector}__slider {
+          background-color: @background-color-on;
+        }
+
+        ~ .@{switch-selector}__slider:before {
```
Comment (antleblanc) [inline packages/oui-switch/_mixins.less]: `~ .@{switch-selector}__slider:before` => `~ .@{switch-selector}__slider::before`

File: packages/oui-switch/_mixins.less [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      top: 0;
+      right: 0;
+      bottom: 0;
+      left: 0;
+      border-radius: ~"calc(@{height} - @{border-width} * 2)";
+      background-color: @background-color-off;
+      box-shadow: inset 0 1px 1px 0 rgba(0, 0, 0, .25);
+      cursor: pointer;
+
+      &:before {
```
Comment (antleblanc) [inline packages/oui-switch/_mixins.less]: `&:before {` => `&::before {`

File: packages/oui-switch/_mixins.less [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        transform: scale(1);
+      }
+    }
+  }
+
+  .switch-square(
+    @switch-selector
+  ) {
+    .@{switch-selector}__slider,
+    .@{switch-selector}__slider:before {
```
Comment (antleblanc) [inline packages/oui-switch/_mixins.less]: `.@{switch-selector}__slider:before` => `.@{switch-selector}__slider::before`

File: packages/oui-switch/_mixins.less [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    @icon-size: @oui-switch-icon-size,
+    @border-width: @oui-switch-border-width
+  ) {
+    position: relative;
+    display: inline-block;
+    width: @width;
+    height: @height;
+
+    &__checkbox {
+      #oui > .visually-hidden();
```
Comment (antleblanc) [inline packages/oui-switch/_mixins.less]: +@import '../oui-accessibility/_mixins'; ?

File: packages/oui-switch/_mixins.less (hunk 1 above)
Comment (CarolineZow) [inline packages/oui-switch/_mixins.less]: Done

File: packages/oui-switch/_mixins.less (hunk 4 above)
Comment (CarolineZow) [inline packages/oui-switch/_mixins.less]: Done

File: packages/oui-switch/_mixins.less (hunk 2 above)
Comment (jimmyfortinx) [inline packages/oui-switch/_mixins.less]: Done

File: packages/oui-switch/_mixins.less (hunk 3 above)
Comment (jimmyfortinx) [inline packages/oui-switch/_mixins.less]: Done

**Pull request merged**